pulsar-broker/src/main/java/org/apache/pulsar/broker/intercept/BrokerInterceptor.java

```diff
@@ -123,6 +123,23 @@ default void messageAcked(ServerCnx cnx, Consumer consumer,
                               CommandAck ackCmd) {
     }
 
+    /**
+     * Intercept when a transaction begins.
+     *
+     * @param tcId Transaction Coordinator Id
+     * @param txnID Transaction ID
+     */
+    default void beginTxn(long tcId, String txnID) {
```

Comment:
It's better to use newTxn to keep consistent with the command.

Reply:
And from the implementation, I think it should be `txnOpened` and `txnEnded`?

Reply:
Done